### PR TITLE
Rewrite main.py, add conftest.py

### PR DESCRIPTION
**Pull request by CapUlDis**
Bro, please, take a look at my changes. I've fixed 'main.py' according to your comments and have created conftest without tests. 'Pytest' command just makes run 'main.py'.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,22 +1,40 @@
...
-app = Flask(__name__)
-FlaskJSON(app)
+def create_app():
+    app = Flask(__name__)
+    return app
 
-@app.route("/")
-def hello():
-    return "Hello World!"
+# if __name__ == '__main__':
```
Comment (nkartashov) [inline main.py]: Don't leave commented code in the PR + it was a good idea to guard with `if __name__ == '__main__':`

File: main.py [hunk 2]
```diff
@@ -1,22 +1,40 @@
...
+app = create_app()
+app.run()
 
 @app.route('/login', methods=['POST'])
 def login():
+    try:
+        if request.headers['Content-Type'] != 'application/json':
+            message = 'Content-Type not json'
```
Comment (nkartashov) [inline main.py]: No need to create additional variables for this, just return inline values.

File: main.py [hunk 3]
```diff
@@ -1,22 +1,40 @@
...
+        if 'name' not in data.keys() or 'password' not in data.keys():
+            message = 'Miss name or/and password'
+            http_code = 400
+            return message, http_code
+        
+        if data['name'] == 'denchik' and data['password'] == 'foobar':
+            message = 'Correct name and password'
+            http_code = 200
+            return message, http_code
+        else:
```
Comment (nkartashov) [inline main.py:28]: No need to have an else clause here as the previous block ends with `return`

File: main.py [hunk 4]
```diff
@@ -1,22 +1,40 @@
...
+    except:
+        message = 'Data has not json format'
+        http_code = 400
+        return message, http_code
     
-    data = request.get_json(force=True)
-
-    if data['name'] == 'denchik' and data['password'] == 'foobar':
-        return 'status = 200 OK'
     else:
```
Comment (nkartashov) [inline main.py]: The `try-else` is run when there was no exception so you can just omit it as `except` ends with `return`

File: main.py [hunk 5]
```diff
@@ -1,22 +1,40 @@
...
-@app.route("/")
-def hello():
-    return "Hello World!"
+# if __name__ == '__main__':
+app = create_app()
+app.run()
 
 @app.route('/login', methods=['POST'])
 def login():
+    try:
```
Comment (nkartashov) [inline main.py]: It's a good idea to limit the scope of `try-except` to only lines which can throw the exception, namely the `json.loads` line.

File: main.py [hunk 6]
```diff
@@ -1,22 +1,40 @@
...
+        if request.headers['Content-Type'] != 'application/json':
+            message = 'Content-Type not json'
+            http_code = 400
+            return message, http_code
+        
+        data = json.loads(request.get_data())
+   
+    except:
```
Comment (nkartashov) [inline main.py]: It's a good idea to specify the exception type here which would also enable you to communicate the error to the user.

File: main.py [hunk 7]
```diff
@@ -1,22 +1,40 @@
...
+        message = 'Data has not json format'
```
Comment (nkartashov) [inline main.py]: "Data is not in json format"

File: main.py [hunk 8]
```diff
@@ -1,22 +1,40 @@
...
-    if data['name'] == 'denchik' and data['password'] == 'foobar':
-        return 'status = 200 OK'
     else:
-        return 'Invalid username/password'
-    
-if __name__ == '__main__':
-    app.run()
\ No newline at end of file
+        if 'name' not in data.keys() or 'password' not in data.keys():
+            message = 'Miss name or/and password'
```
Comment (nkartashov) [inline main.py]: "Name or/and password are missing"

**Review by nkartashov — CHANGES_REQUESTED** [review]

File: main.py (hunk 1 above)
Comment (CapUlDis) [reply main.py]: But with `if __name__ == '__main__':` pytext can't run app and tests.

File: main.py (hunk 6 above)
Comment (CapUlDis) [reply main.py]: I've tried to put here `except JSONDecodeError as err:`, but it raised Error:  'JSONDecodeError is not defined'. But when I put not a json data without exception, program raised JSONDecodeError

File: main.py (hunk 7 above)
Comment (CapUlDis) [reply main.py]: Done

File: main.py (hunk 8 above)
Comment (CapUlDis) [reply main.py]: Done

File: main.py (hunk 6 above)
Comment (CapUlDis) [reply main.py]: Dealt with it.

File: main.py (hunk 5 above)
Comment (CapUlDis) [reply main.py]: Done!

File: main.py (hunk 4 above)
Comment (CapUlDis) [reply main.py]: Done!

File: main.py (hunk 3 above)
Comment (CapUlDis) [reply main.py:28]: Don't understand why. Here we have case, when server gets right name and password, and case, when he gets wrong name, password. In both cases he should return messages. What's wrong?

File: main.py (hunk 2 above)
Comment (CapUlDis) [reply main.py]: Done!

**Pull request closed without merging**